themes/default/content/what-is/what-is-configuration-management.md

```diff
@@ -0,0 +1,73 @@
+
+## What are some common configuration management tools?
+
+There are a number of configuration management tools on the market, but here are a few of the most commonly mentioned tools:
+
+* Ansible
+* Chef
+* Puppet
+* Salt
+
+Although the specific implementation details vary from tool to tool---some of thse tools use YAML while others use a domain-specific language (DSL)---there are a few common attributes that most configuration management tools share:
+
+* **Declarative configuration**: Most configuration management tools follow a declarative approach where the user specifies the desired state, and the tool takes care of reaching that state.
+* **Idempotence**: Configuration management tools ensure idempotence, meaning that applying the same configuration multiple times yields the same result, promoting reliability.
+* **Automation**: Automation is a fundamental attribute of configuration management tools, allowing the tools to execute tasks consistently and reduce the risk of human error.
+* **Stateless**: Configuration management tools typically do not maintain any state, or saved information, on the configuration status of systems being managed.
+
+## What are some benefits of configuration management?
+
+Configuration management offers a number of tangible benefits:
+
+* **Consistency**: By defining the desired state and then _identifying_ and _remediating_ the specific things do not align with the desired state, configuration management helps bring greater consistency to systems. For example, configuration management helps ensure that the same version of a particular software package is installed across a specific subset of operating systems instances, or that all operating systems instances are patched with a particular security update.
+* **Reliability**: Inconsistency is the bane of reliability. _Snowflake servers_ that have manual tweaks, incorrect hotfixes, or missing software updates make it harder to fix issues. Automation helps reduce inconsistency by eliminating human error and variance in manually-configured systems.
+* **Scalability:** Defined configurations that can be deployed via configuration management tool make it possible to scale systems faster, easier, and more reliably.
+* **Accountability**: Similar to infrastructure as code, configuration management enables the desired state to be checked into version control. Changes to the desired state can be tracked, providing accountability and visibility.
+
+## What are some practical use cases for configuration management?
+
+Configuration management has a number of practical use cases for many organizations:
+
+* Automated configuration of new systems (typically servers, but could also apply to network devices or other systems)
+* Recovery from outages
+* Remediation of security flaws
+* Patching or updating software or operating systems
+* Deployment of software packages or applications
+
+## Are there any disadvantages to configuration management?
```

Comment:
i wonder if another disadvantage is its is "operations based" in that you apply a set of rules to get to desired state. at the end of the day, there can be enough drift that desired state never gets achieved. 

the alternative is just having entire environments configured from scratch and that is where you get the AMI based approach of AWS or Docker containers. wondering how we should work that into here.

Reply:
I've added some text immediately before this heading (around line 56-57) mentioning immutable infrastructure (AMI-based approach) and containers. Good call.